shared-bindings/microcontroller/__init__.c

```diff
@@ -113,12 +114,13 @@ const mp_obj_module_t mcu_pin_module = {
 STATIC const mp_rom_map_elem_t mcu_module_globals_table[] = {
     { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_microcontroller) },
     { MP_ROM_QSTR(MP_QSTR_delay_us), MP_ROM_PTR(&mcu_delay_us_obj) },
+    { MP_ROM_QSTR(MP_QSTR_core),  MP_ROM_PTR(&mcu_core_obj) },
```

Comment:
This will break the ESP8266. You probably want to #ifdef it out like nvm.

Reply:
Instead, I implemented `core` for ESP8266. `core.temperature` does not exist (or at least there's no public API yet) and will return `None`. The `common-hal` temperature function returns `NAN` to signal no value is available. `core.frequency` is available and I return that.